src/main/java/seedu/address/model/person/Person.java

```diff
@@ -41,6 +42,23 @@ public Person(Name name, Phone phone, Email email, Address address, Set<Tag> tag
         this.uuid = UUID.randomUUID();
     }
 
+    /**
+     * Creates a new Person instance by cloning the details of an existing person.
+     * This constructor is used for creating a copy of {@code personToEdit} with the intention
+     * of editing or updating its attributes. It copies all the attributes of the provided person,
+     * including name, phone, email, address, tags, and UUID, into a new Person object.
+     *
+     * @param personToEdit The person whose details are to be cloned into this new instance.
+     */
+    public Person(Person personToEdit) {
+        // Clone existing fields
+        this.name = personToEdit.name;
+        this.phone = personToEdit.phone;
+        this.email = personToEdit.email;
+        this.address = personToEdit.address;
+        this.tags = new HashSet<>(personToEdit.tags);
+        this.uuid = personToEdit.uuid;
+    }
```

Comment:
Cloning a person is handled by PersonBuilder class. Is the intent of bringing this functionality into Person intended to deprecate PersonBuilder?